src/dtx/dtx_internal.h

```diff
@@ -94,6 +94,21 @@ extern uint32_t dtx_agg_thd_cnt_lo;
 #define DTX_AGG_THD_AGE_MIN	210
 #define DTX_AGG_THD_AGE_DEF	630
 
+/* There is race between DTX aggregation and DTX refresh. Consider the following scenario:
+ *
+ * The DTX leader triggers DTX commit for some DTX entry, then related DTX participants
+ * (including the leader itself) will commit the DTX entry on each own target in parallel.
+ * It is possible that the leader has already committed locally but DTX aggregation removed
+ * the committed DTX very shortly after the commit. On the other hand, on some non-leader
+ * before the local commit, someone triggers DTX refresh for such DTX on such non-leader.
+ * Unfortunately the DTX entry has already gone on the leader. Then the non-leader will
+ * get -DER_TX_UNCERTAIN, that will cause related application to fail unexpectedly.
+ *
+ * So even if the system has DRAM pressure, we still need to keep some very recent committed
+ * DTX entries to handle above race.
+ */
+#define DTX_AGG_AGE_PRESERVE	3
```

Comment:
Even with this extra delay, refresh still could happen after the entry being aggregated on leader, right? How will we handle this? Is this a temporary workaround or final solution? What about aggregating only globally committed entry on leader?

Reply:
This extra delay is for the case of DTX commit difference on related targets, usually, as long as no target corruption, 3 seconds is enough for DTX committed on all related targets. After the global DTX commit, there will be no DTX refresh unless some target missed the DTX commit RPC. Under such case, if the committed DTX entry is removed by DTX aggregation, then the UNCERTAIN issue will happen again.

Make the DTX aggregation globally can resolve the issue, but the cost  may be too much, it needs additional RPCs. So we do not make such decision yet.